source/addonStore/network.py

```diff
@@ -9,6 +9,7 @@
 )
 import os
 import pathlib
+import posixpath
```

Comment:
I can't find any documentation for this in the python docs, where is this coming from?

perhaps we could just strip any trailing slashes as an equivalent?
or https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

Reply:
> I can't find any documentation for this in the python docs, where is this coming from?

It is the Posix-specific version of `os.path`. Per [the `os.path` docs](https://docs.python.org/3/library/os.path.html):

> **Note:** Since different operating systems have different path name conventions, there are several versions of this module in the standard library. The [`os.path`](https://docs.python.org/3/library/os.path.html#module-os.path "os.path: Operations on pathnames.") module is always the path module suitable for the operating system Python is running on, and therefore usable for local paths. However, you can also import and use the individual modules if you want to manipulate a path that is *always* in one of the different formats. They all have the same interface:
 >
> - `posixpath` for UNIX-style paths
> - `ntpath` for Windows paths


> perhaps we could just strip any trailing slashes as an equivalent?

While this may work, I suspect `os.path.join` handles a lot of edge cases that we aren't thinking of. If you think this is unimportant, I can make the switch though.

> Or https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

I initially didn't use this as the behaviour varies a fair bit with the location of slashes in the url (so is `posixpath.join`). However, as long as we normalise the base URL to have a trailing slash I think this should be fine. It's also somewhat more self-documenting.